operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Context.java

```diff
@@ -22,4 +25,21 @@ default <T> Optional<T> getSecondaryResource(Class<T> expectedType) {
   ControllerConfiguration<P> getControllerConfiguration();
 
   ManagedDependentResourceContext managedDependentResourceContext();
+
+  ResourceID currentlyReconciledResourceID();
```

Comment:
Why are we exposing this to the user? This is not usable at all for him. Could be very confusing? Is this this for metrics?